thorium/fields.py

```diff
@@ -33,8 +33,9 @@ def __init__(self, default=NotSet, notnull=False, readonly=False,
 
         # create validator
         self._validator = self._get_validator()
+        if callable(default):
```

Comment:
This isn't going to work, the __init__ here runs when the server starts up, not when the Resource is created